handlers/photo-audio-haptics-handler/src/server.ts

```diff
@@ -126,15 +130,22 @@ app.post("/handler", async (req, res) => {
         const [ttsInfo, geometryInfo] = utils.generateObjDet(preObjDet, preGroupData);
         ttsData.push(...ttsInfo);
         objGeometryData.push(...geometryInfo);
+        
     }
-
+    
     // Concatenate adjacent text entries
     for (let i = 0; i < ttsData.length - 1; i++) {
         if (ttsData[i].type === "text" && ttsData[i + 1].type === "text") {
             ttsData[i].value += " " + ttsData[i + 1].value;
             ttsData.splice(i + 1, 1);
         }
     }
+    if (preSemSeg) {
```

Comment:
Since this is just added to the end, then you're adding functionality in the browser to skip to this part when necessary, right?